### PR TITLE
Bolt slack bot prototype

### PR DESCRIPTION
**Pull request by Zhornovyi**
### 🛠 Changes being made

1. Updated the structure of the slack bock package
2. Enabled modification of the command: 
`/generate_job_description https://chatcraft.org/c/iramykytyn/LkjsdteMwbbdlRkJoccI_ ` - update the command url
`/generate_job_description Hint: new hints to use ` - update the command hint
3. Updated requirements
4. Switched to `slack_bolt` library (It has is more high level sdk from slack: [ref](https://api.slack.com/start/apps#tools))
5. Updated dockerfile 
6. Added ngrok setup + automatic updating the app manifest with the public provided by the ngrok
7. removed `/get_chats` command
8. Added App Home view, with basic information (as for me command now is not really useful, we can add just  one for configuration using provided file)
Знімок екрана 2023-08-30 о 23 37 20

### 🧪 How was it tested? / Demo

⚠️ No unit test yet - should be added later. 

### 🏎 Quality check

- [x]  Set base branch to main.
- [x]  Pull request is prepared: Added to sprint, added yourself as assignee, reviewers are added.
- [ ]  PR is not very big (about 200 lines of change ideally).
- [ ]  I have added comments to my code, particularly in hard-to-understand areas.
- [x]  Doesn't create any new FIXME or TODO comments. If not, please explain.
- [x]  Tested manually (explained the scenario, added screenshots)
- [x]  I have tested my changes locally before pushing them to this branch.
- [ ]  I have updated the documentation accordingly.
- [ ]  Followed best practice rules:https://google.github.io/styleguide/pyguide.html

### CODE REVIEW

File: slack_bot/main.py
Comment (iramykytyn) [inline slack_bot/main.py:13]: So there is not slash commands anymore and everything will be handled as events, right?
I will need to update it after #20 i merged.

File: slack_bot/run_ngrok.py
Comment (iramykytyn) [inline slack_bot/run_ngrok.py]: so will it update url in Slack app automatically? This is cool if so!

**Review by iramykytyn — APPROVED** [review]
LGTM! But I will merge it later after #20 since I will need to update it and fix merge conflicts accordingly.

File: slack_bot/run_ngrok.py
Comment (Zhornovyi) [reply slack_bot/run_ngrok.py]: yes, the only problem is in expiring app config token, it should be updated every 12 hours. So, I think we should have a static public url anyway.

File: slack_bot/main.py
Comment (Zhornovyi) [reply slack_bot/main.py:13]: kind of, earlier we had `/process_command` endpoint that handled all commands, now I've just changed commands request url to events endpoint and let Bolt handle it.

File: slack_bot/run_ngrok.py
Comment (iramykytyn) [reply slack_bot/run_ngrok.py]: can't we just use a refresh token to make it last forever?

Comment (iramykytyn) [reply slack_bot/run_ngrok.py]: If I understand teh problem correctly then implementing refrech logic should fix it.
Screenshot 2023-08-31 at 12 40 09

Comment (Zhornovyi) [reply slack_bot/run_ngrok.py]: sure, we can, I didn't do that cause I thought ngrok is temporary solution, and we will have static public url  later.

**Comment by iramykytyn** [comment]
I was able to run it, but I needed to do some  ngrok configuration on my mashine, so I added more log messages to simplify debugging in the future if issue occur.

I think I will better merge it now and finish #20 on the base of it.

File: slack_bot/run_ngrok.py
Comment (iramykytyn) [reply slack_bot/run_ngrok.py]: I don't see how ngrok and access tokens are related, could you please explain? This token will expire anyway, won't it?

**Pull request merged**
